pkgs/by-name/in/intel-oneapi/base.nix

```diff
@@ -0,0 +1,254 @@
+{
+  lib,
+  fetchurl,
+  zlib,
+  rdma-core,
+  libpsm2,
+  ucx,
+  numactl,
+  level-zero,
+  pkg-config,
+  libdrm,
+  elfutils,
+  xorg,
+  glib,
+  nss,
+  nspr,
+  dbus,
+  at-spi2-atk,
+  cups,
+  gtk3,
+  pango,
+  cairo,
+  mesa,
+  expat,
+  libxkbcommon,
+  eudev,
+  alsa-lib,
+  ncurses5,
+  bzip2,
+  gdbm,
+  libxcrypt-legacy,
+  freetype,
+  gtk2,
+  gdk-pixbuf,
+  fontconfig,
+  libuuid,
+  sqlite,
+
+  # The list of components to install;
+  # Either [ "all" ], [ "default" ], or a custom list of components.
+  # If you want to install all default components plus an extra one, pass [ "default" <your extra components here> ]
+  # Note that changing this will also change the `buildInputs` of the derivation.
+  # The default value is not "default" because some of the components in the defualt set are currently broken.
+  components ? [
+    "intel.oneapi.lin.advisor"
+    "intel.oneapi.lin.dpcpp-cpp-compiler"
+    "intel.oneapi.lin.dpcpp_dbg"
+    "intel.oneapi.lin.vtune"
+    "intel.oneapi.lin.mkl.devel"
+  ],
+
+  intel-oneapi,
+
+  # For tests
+  runCommand,
+  libffi,
+  stdenv,
+}:
+intel-oneapi.mkIntelOneApi rec {
```

Comment:
`rec` is an anti-pattern and breaks overriding attributes; `mkIntelOneApi` should support a fixed-point argument.